### PR TITLE
[NVIDIA] Enable CuTe DSL BF16 GEMM on SM107

### PR DESCRIPTION
**Pull request by YAMY1234**
## Motivation

The CuTe DSL BF16 GEMM backend is selected on SM10x GPUs, but both runtime entry points reject SM107. Enabling the architecture gate also exposes a bias conversion issue: DLPack rejects a bias view that retains `requires_grad=True`, even during inference under `torch.no_grad()`.

## Modifications

- Allow SM107 in both CuTe DSL BF16 GEMM runtime entry points.
- Detach the read-only bias view before exporting it through DLPack. This does not add a copy or synchronization.
- Update backend diagnostics and CLI help to describe SM10x support.
- Extend the existing kernel test with an SM107 workload shape and a `requires_grad=True` bias under `torch.no_grad()`.

The GEMM kernel math, tactic selection, and launch behavior are unchanged.

## Accuracy Tests

The existing numerical comparison test now covers the added shape and gradient-bearing bias case. Controlled SM107 validation passed for eager execution, CUDA Graph execution, and an end-to-end accuracy workload on the pinned development base.

Current upstream-main GPU CI is pending while this PR remains a draft.

## Speed Tests and Profiling

Controlled end-to-end SM107 testing confirmed that enabling this backend removes the fallback to `torch.nn.functional.linear`. No kernel implementation or tactic was changed.

Current upstream-main performance confirmation is pending while this PR remains a draft.

## Checklist

- [x] Format the code with pre-commit.
- [x] Extend the existing unit test.
- [x] Update user-facing backend documentation.
- [ ] Complete GPU CI on current upstream main.
- [ ] Add current-main performance confirmation.

## Local Validation

- `pre-commit run --files` passed for all four modified files.
- `git diff --check` passed.
- Python compilation passed for all four modified files.

---
### CI States

Latest PR Test (Base): :x: [Run #31033970223](https://github.com/sgl-project/sglang/actions/runs/31033970223)
Latest PR Test (Extra): :x: [Run #31033969425](https://github.com/sgl-project/sglang/actions/runs/31033969425)

### CODE REVIEW

File: python/sglang/kernels/ops/gemm/cutedsl_bf16_gemm.py [hunk 1]
```diff
@@ -1352,8 +1354,8 @@ def _tgv_bf16_gemm_run(
     weight: torch.Tensor,
     bias: Optional[torch.Tensor],
 ) -> torch.Tensor:
-    if get_device_sm() not in (100, 103):
```
Comment (b8zhong) [inline python/sglang/kernels/ops/gemm/cutedsl_bf16_gemm.py:1355]: Can we just change is to is_sm100_supported?

Comment (leejnau) [reply python/sglang/kernels/ops/gemm/cutedsl_bf16_gemm.py:1355]: maybe `is_sm10x_supported` ?   what is Supports 110 exactly?

Comment (mmangkad) [reply python/sglang/kernels/ops/gemm/cutedsl_bf16_gemm.py:1355]: `is_sm10x_supported` would’ve been nice if there were no need to check different cuda versions, but since sm107 needs >= 13.4 while `is_sm100_supported` currently checks >= 12.8 (actually, sm103 needs >= 12.9), maybe for clarity rename it to `is_sm100_or_sm103_supported` and add `is_sm107_supported`

Comment (b8zhong) [reply python/sglang/kernels/ops/gemm/cutedsl_bf16_gemm.py:1355]: @leejnau You can ignore the other part of my comment. Was typing something else on another tab or something..

**Comment by leejnau** [comment]
I tested Kimi K3 on sm107 and the cutedsl bf16 gemm works ✅

**Review by leejnau — APPROVED** [review]

File: python/sglang/kernels/ops/gemm/cutedsl_bf16_gemm.py (hunk 1 above)
Comment (b8zhong) [reply python/sglang/kernels/ops/gemm/cutedsl_bf16_gemm.py:1355]: @mmangkad I don't think we really need to check the CUDA version. It should be correct already (@leejnau What do you think?)

Comment (YAMY1234) [reply python/sglang/kernels/ops/gemm/cutedsl_bf16_gemm.py:1355]: Thanks! I updated the PR to use is_sm100_supported() instead of explicitly checking for SM100/103/107.

BTW, Lee and I talked about it, and we don’t think we need a separate `is_sm107_supported()` for now. If we add features that are only SM107-specific later, we can add it then as a follow up

**Review by b8zhong — APPROVED** [review]

File: python/sglang/kernels/ops/gemm/cutedsl_bf16_gemm.py (hunk 1 above)
Comment (mmangkad) [reply python/sglang/kernels/ops/gemm/cutedsl_bf16_gemm.py:1355]: Yeah maybe not for now, but at some point we'll probably have to distinguish them

**Review by mmangkad — APPROVED** [review]
